kore/src/Kore/Step/Simplification/Simplify.hs

```diff
@@ -154,7 +155,8 @@ class (WithLog LogMessage m, MonadSMT m, MonadProfiler m)
 
     simplifyCondition
         :: InternalVariable variable
-        => SideCondition variable
+        => NotSimplifier m2
```

Comment:
We don't want to add this argument here, because how the instance of `MonadSimplify` chooses to simplify `\not` is an implementation detail: it could do anything or nothing at all, and we don't want that to leak out into the interface.